sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,601 @@
+        if not _check_soft_dependencies("torch", severity="none"):
+            raise ModuleNotFoundError(
+                "torch is a required optional dependency for RBFForecaster. "
+                "Please install torch to use this forecaster. "
+                "Instructions can be found at https://pytorch.org/get-started/locally/"
+            )
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the RBF-based model to the provided time series data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            The time series data to be fitted.
+        X : optional
+            Additional exogenous data (not used).
+        fh : optional
+            Forecasting horizon (not used).
+        """
+        import torch
+        from torch.utils.data import DataLoader, TensorDataset
+
+        LazyLinear = self._create_lazy_linear()
+        RBFLayer = self._create_rbf_layer()
+        RBFNetwork = self._create_rbf_network(LazyLinear, RBFLayer)
+
+        self._y = y.copy()
+
+        if isinstance(y, pd.Series):
+            y_scaled = self.scaler.fit_transform(y.values.reshape(-1, 1))
+        else:
+            y_scaled = self.scaler.fit_transform(y.values)
+
+        X_train, y_train = self._create_windows(y_scaled)
+
+        if len(X_train) == 0:
+            raise ValueError(
+                f"Not enough samples to create training windows. "
+                f"Try reducing window_length (currently {self.window_length}) "
+                f"or stride (currently {self.stride})."
+            )
+
+        X_tensor = torch.FloatTensor(X_train).to(self.device)
+        y_tensor = torch.FloatTensor(y_train).to(self.device)
+
+        dataset = TensorDataset(X_tensor, y_tensor)
+        dataloader = DataLoader(
+            dataset, batch_size=min(self.batch_size, len(X_train)), shuffle=True
+        )
+
+        input_size = self.window_length
+        output_size = 1
+
+        self.model = RBFNetwork(
+            input_size=input_size,
+            hidden_size=self.hidden_size,
+            output_size=output_size,
+            centers=self.centers,
+            gamma=self.gamma,
+            rbf_type=self.rbf_type,
+            linear_layers=self.linear_layers,
+        ).to(self.device)
+
+        criterion = self._create_loss_fn()
+
+        self.model.train()
+        optimizer = None
+
+        for epoch in range(self.epochs):
+            total_loss = 0
+            for batch_idx, (batch_X, batch_y) in enumerate(dataloader):
+                if optimizer is None:
+                    _ = self.model(batch_X)
+                    optimizer = self._create_optimizer()
+
+                optimizer.zero_grad()
+                outputs = self.model(batch_X)
+                loss = criterion(outputs, batch_y)
+                loss.backward()
+                optimizer.step()
+                total_loss += loss.item()
+
+        self._is_fitted = True
```

Comment:
not necessary to do that, the boilerplate layer (`fit`) takes care of state handling.